cli/command/stack/kubernetes/loader.go

```diff
@@ -1,18 +1,34 @@
 package kubernetes
 
 import (
+	"github.com/docker/cli/cli/compose/loader"
+	"github.com/docker/cli/cli/compose/schema"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	apiv1beta1 "github.com/docker/cli/kubernetes/compose/v1beta1"
+	"github.com/pkg/errors"
 	yaml "gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const v1beta1MaxComposeVersion = "3.5"
+
 // LoadStack loads a stack from a Compose config, with a given name.
 func LoadStack(name string, cfg composetypes.Config) (*apiv1beta1.Stack, error) {
+	cfg.Version = v1beta1MaxComposeVersion
 	res, err := yaml.Marshal(cfg)
 	if err != nil {
 		return nil, err
 	}
+
+	// reload the result to check that it produced a valid 3.5 compose file
+	resparsedConfig, err := loader.ParseYAML(res)
+	if err != nil {
+		return nil, err
+	}
+	if err = schema.Validate(resparsedConfig, v1beta1MaxComposeVersion); err != nil {
+		return nil, errors.Wrapf(err, "the generated compose yaml file is invalid with v%s", v1beta1MaxComposeVersion)
```

Comment:
Remove "generated" in the message? I don't know if it will be a useful information for the user?